src/hlstrafe.cpp

```diff
@@ -1214,18 +1214,26 @@ namespace HLStrafe
 		}
 	}
 
-	PositionType GetPositionType(PlayerData& player, TraceFunc traceFunc)
+	PositionType GetPositionType(PlayerData& player, TraceFunc traceFunc, PointContentsFunc pointContentsFunc)
 	{
 		// Check water. If we're under water, return here.
-		if (player.WaterLevel > 1)
+		const int CONTENTS_WATER = -3;
+		const int CONTENTS_TRANSLUCENT = -15;
+
+		float point[3];
+		VecCopy<float, 3>(player.Origin, point);
+
+		point[2] += (player.Ducking ? (VEC_DUCK_HULL_MIN[2] + VEC_DUCK_HULL_MAX[2]) : (VEC_HULL_MIN[2] + VEC_HULL_MAX[2])) / 2.f;
```

Comment:
This does not seem to match what `PM_CheckWater` does: https://github.dev/ValveSoftware/halflife/blob/c7240b965743a53a29491dd49320c88eecf6257b/pm_shared/pm_shared.c#L1480

Reply:
I'm intentionally skipping checking for waterlevel == 1 and check for waterlevel > 1, which is a hull midpoint, since i thought that's what we care about https://github.dev/ValveSoftware/halflife/blob/c7240b965743a53a29491dd49320c88eecf6257b/pm_shared/pm_shared.c#L1499-L1503. The expression is reduced to `point[2] += 0.0` anyway, i just wrote it in hopes that the compiler would optimize it away.

Reply:
The check for waterlevel == 1 is still needed in case the water brush is floating in the air. We still want correct behavior in this case. And I'd leave the x and y hull calculations in-tact, even if the constants used for HL1 result in zeros. Easier to check and to adjust to different conditions if that's ever needed.